### ISSUE TITLE
Alternator test failing in Scylla 4.0 on s390x 

### ISSUE THREAD

**Issue by nyh** [scylladb/scylla#7008]
Prasanna Kelkar reported on the scylladb-dev mailing list that:

"_I was able to build v4.0.3 on s390x architecture with similar steps. But while running the tests observed one test case failure for alternator/run. Test case alternator/run failed due to segmentation fault with following logs._
```
...
seastar::future<> seastar::futurize<seastar::future<> >::apply<seastar::httpd::connection::read_one()::{lambda()#1}>(seastar::httpd::connection::read_one()::{lambda()#1}&&, std::tuple<>&&) [clone .isra.551] at /usr/local/include/fmt/format.h:645
...
```
"

It turns out that this is a use-after-move bug in **Seastar** that @espindola has already fixed back in May, in commit c8c209da30190aaf63f6055801a35dcda37c05e3 of Seastar. This was part of a series of potential bug fixes that @espindola discovered by using the sanitizer in clang, but were never seen to cause actual bugs in the wild - until now.

The patches of that series are already included in Scylla 4.1, but need backporting to Scylla 4.0 (which the reporting user used). I'll do this backport now.

**Comment by nyh** [comment]
Backported the relevant Seastar patches to 4.0, now in next-4.0 commit 1d3c00572cb8395bde03b3a42afcd9c9c6d3d953
Issue should hopefully be auto-closed when next-4.0 is promoted to branch-4.0.

**Comment by nyh** [comment]
This was never auto-closed, closing manually.
